drivers/soundwire/intel.c

```diff
@@ -1130,11 +1130,24 @@ static void *intel_get_sdw_stream(struct snd_soc_dai *dai,
 	return dma->stream;
 }
 
+/* handle SUSPEND trigger to free the DAI widget during suspend */
+static int intel_trigger(struct snd_pcm_substream *substream, int cmd, struct snd_soc_dai *dai)
+{
+	struct sdw_cdns *cdns = snd_soc_dai_get_drvdata(dai);
+	struct sdw_intel *sdw = cdns_to_intel(cdns);
+
+	if (cmd == SNDRV_PCM_TRIGGER_SUSPEND || cmd == SNDRV_PCM_TRIGGER_STOP)
+		return intel_free_stream(sdw, substream, dai, sdw->instance);
```

Comment:
this also doesn't look good since we already call intel_free_stream() from the hw_free() callback, so in the regular case we'd seen two FREE IPCs. that's not so good.